### PR TITLE
Allow disabling rustfmt with a dir-local/file-local variable

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

File: rust-mode.el [hunk 1]
```diff
@@ -1317,9 +1318,11 @@ This is written mainly to be used as `end-of-defun-function' for Rust."
   (setq-local parse-sexp-lookup-properties t)
   (setq-local electric-pair-inhibit-predicate 'rust-electric-pair-inhibit-predicate-wrap)
   (add-hook 'after-revert-hook 'rust--after-revert-hook 'LOCAL)
```
Comment (Stebalien) [inline rust-mode.el]: Unrelated but this looks wrong. Shouldn't this end with `nil t` (or `nil 'LOCAL`), not just `'LOCAL`?

Comment (tomjakubowski) [inline rust-mode.el]: Yeah, that sounds right to me.

File: rust-mode.el [hunk 2]
```diff
@@ -186,6 +186,7 @@ function or trait.  When nil, where will be aligned with fn or trait."
   "Format future rust buffers before saving using rustfmt."
   :type 'boolean
   :safe #'booleanp)
+(put 'rust-format-on-save 'safe-local-variable 'booleanp)
```
Comment (mrBliss) [inline rust-mode.el]: Isn't this line redundant given that the line above expresses exactly the same thing (in a nicer way)?

`defcustom`'s documentation says:

```
...
:safe	Set SYMBOL's `safe-local-variable' property to VALUE.
        See Info node `(elisp) File Local Variables'.
...
```

Comment (Stebalien) [inline rust-mode.el]: Didn't know that... thanks!

**Comment by Stebalien** [comment]
I ended up just tacking on the second bug fix. I can split it into two pull requests if you'd like but I did it this way so I wouldn't forget.

File: rust-mode.el (hunk 1 above)
Comment (MicahChalmer) [inline rust-mode.el]: Yup - that was a mistake.  Thanks for catching it.  Fortunately the hook would usually be fairly harmless when run in other modes...but much better if it's local as intended.

**Pull request merged**